src/components/include/cbuf.h

```diff
@@ -332,66 +297,37 @@ cbufp2buf(cbufp_t cb, int len) { return __cbuf2buf((cbuf_t)cb, len, 0); }
  * or the component can call cbufp_free.
  */
 static inline void
-__cbufp_send(cbuf_t cb, int free)
+__cbuf_send(cbuf_t cb, int free)
 {
 	u32_t id;
 	struct cbuf_meta *cm;
-	int tmem;
-
-	cbuf_unpack(cb, &id, &tmem);
-	assert(tmem == 0);
 
-	CBUF_TAKE();
-	cm = cbuf_vect_lookup_addr(cbid_to_meta_idx(id), 0);
+	cbuf_unpack(cb, &id);
 
-	assert(cm && cm->nfo.v);
-	assert(cm->nfo.c.refcnt);
-	assert(!(cm->nfo.c.flags & CBUFM_TMEM));
-	assert(cm->owner_nfo.c.nsent < TMEM_SENDRECV_MAX);
+	cm = cbuf_vect_lookup_addr(cbid_to_meta_idx(id));
+	assert(cm && cm->nfo);
+	assert(CBUFM_GET_REFCNT(cm));
+	assert(cm->owner_nfo.c.nsent < CBUF_SENDRECV_MAX);
 
 	cm->owner_nfo.c.nsent++;
-	if (free) cm->nfo.c.refcnt--;
+	if (free) CBUFM_DEC_REFCNT(cm);
 	/* really should deal with this case correctly */
-	assert(!(cm->nfo.c.flags & CBUFM_RELINQ)); 
-	CBUF_RELEASE();
+	assert(!CBUF_RELINQ(cm));     //???
 }
 
 static inline void
-cbufp_send_deref(cbufp_t cb)
-{ __cbufp_send(cb, 1); }
+cbuf_send_deref(cbuf_t cb)
+{ __cbuf_send(cb, 1); }
 
 static inline void
-cbufp_send(cbufp_t cb) 
-{ __cbufp_send(cb, 0); }
+cbuf_send(cbuf_t cb) 
+{ __cbuf_send(cb, 0); }
 
-extern cvect_t alloc_descs; 
-struct cbuf_alloc_desc {
-	int cbid, length, tmem;
-	void *addr;
-	struct cbuf_meta *meta;
-	struct cbuf_alloc_desc *next, *prev, *flhead; /* freelist */
+struct cbuf_freelist {
+	struct cbuf_meta freelist_head[64];    //???
```

Comment:
`64` should be a macro.  Is 64 chosen for any reason?  Are these power of 2 freelists?  If they are, we're maxing out at a ridiculously large number of pages (2^64 pages) which is completely unnecessary.